Automattic/jetpack
# Social: Register meta based on feature availablility

To prevent post meta from being added to all posts when SIG or enhanced publishing is not available, we should only register the metadata and set the defaults, when the features are part of the current plan.





Fixes #

## Proposed changes:
This change checks the plan information and if the Social features aren't available then it doesn't register the associated metadata

### Other information:


## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
Apologies for all the manual testing!

* With a test site enable auto-sharing, connect an account and publish a post.
* Check the metadata associated with the post, locally and after syncing. There should be no `jetpack_social_options`
* Buy the Advanced plan
* Publish more posts, both with and without attached media.
* Again check the metadata and ensure that `jetpack_social_options` is only associated with the post where there is attached media
* Add the blog sticker to enable SIG. You may need to run `do_action('jetpack_heartbeat')` in `jetpack docker wp shell` to update the plan info.
* Publish more posts and check that the metadata now includes the `social_image_generator` property in the `jetpack_social_options`.
